pkg/api/kubernetesagentpool/validate.go

```diff
@@ -0,0 +1,99 @@
+package kubernetesagentpool
+
+import (
+	"fmt"
+	"strings"
+)
+
+func (p *AgentPool) Validate() error {
+
+	// -------------------------------
+	// Kubernetes Version
+	if p.Properties.KubernetesVersion == "" {
+		return fmt.Errorf("Empty Kubernetes version")
+	}
+	if !strings.Contains(p.Properties.KubernetesVersion, ".") {
+		return fmt.Errorf("Invalid Kubernetes version")
+	}
+
+	// -------------------------------
+	// Kubernetes Version
+	if p.Properties.KubernetesEndpoint == "" {
+		return fmt.Errorf("Empty Kubernetes endpoint")
+	}
+	if !strings.Contains(p.Properties.KubernetesVersion, ".") {
+		return fmt.Errorf("Invalid Kubernetes endpoint")
+	}
+
+	// -------------------------------
+	// Agent pools
+	if len(p.Properties.AgentPoolProfiles) < 1 {
+		return fmt.Errorf("Missing agent pool definition")
+	}
+	for _, agentPool := range p.Properties.AgentPoolProfiles {
+		if agentPool.Count < 1 {
+			return fmt.Errorf("Must using a value greater than 0 for agent pool count")
+		}
+		if agentPool.OSDiskSizeGb != 0 && agentPool.OSDiskSizeGb < 2 {
+			return fmt.Errorf("Must use a disk size greater than 2gb for agent pool")
+		}
+		if agentPool.Name == "" {
+			return fmt.Errorf("Empty name for agent pool")
+		}
+	}
+
+	// -------------------------------
+	// Network Profile
+	if p.Properties.NetworkProfile.PodCIDR == "" {
+		return fmt.Errorf("Empty pod CIDR in network profile")
+	}
+	if p.Properties.NetworkProfile.ServiceCIDR == "" {
+		return fmt.Errorf("Empty service CIDR in network profile")
+	}
+
+	// -------------------------------
+	// Certificate Profile
+	if p.Properties.CertificateProfile.APIServerCertificate == "" {
+		return fmt.Errorf("Empty API certificate")
```

Comment:
I mean the model only has keys, and certs for the API server.. it's a nit but happy to add it